### ISSUE TITLE
Telemetry counts for Search Access Point

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/fenix#6556]
Need to verify that we're correctly collecting searches from Fenix's Search Access Points (SAP) (like urlbar, search engine, etc). Here is the list of [Desktop's access points](https://searchfox.org/mozilla-central/source/browser/modules/BrowserUsageTelemetry.jsm#63) to compare to.

### Acceptance Criteria
- [x] Telemetry type ~~created that is similar to the [desktop SEARCH_COUNTS histogram](https://searchfox.org/mozilla-central/source/toolkit/components/telemetry/Histograms.json#8950), or a similar data format cleared by Frank Bertsch (Data Engineering)~~ This is `labeled_counter`.
- [ ] Data review for any additional SAP (here is [Desktop data review](https://bugzilla.mozilla.org/show_bug.cgi?id=1499193#c11))
- [ ] Search access points telemetry has parity (if possible) with Desktop, and collected as the format `<engine-name>.<search-access-point>` or `<engine-name>.alias` for search shortcuts
    - For instance, the histogram key for a Google search via the urlbar might look like `google-2018.urlbar`
    - Desktop's `recordSearch` [implementation](https://searchfox.org/mozilla-central/source/browser/modules/BrowserUsageTelemetry.jsm#415)
- [ ] verify w/ @mconnormoz from the search team about whether Fenix's search access points are sufficient to have parity w/ desktop

**Comment by ValentinTimisica** [comment]
Hello @liuche, @mconnormoz. I investigated this and saw that we could not reach SAP parity with desktop because atm most of the search entries do not exist on Fenix.
- `abouthome`  - as a SAP for searching from the HomeFragment is not possible.
- `contextmenu` - will be implemented in the future in #5695.
- `newtab` - not sure what SAP this key should be mapped to.
- `searchbar` - on Fenix we have the `BrowserToolbar` for which we can track searches with SAP.
**This would be the only SAP common with desktop atm**.
- `system` - not sure what SAP this key should be mapped to.
- `urlbar` - other than `BrowserToolbar` we don't have other components which can be mapped with this SAP.
- `webextension` - will be implemented in the future in #5315.

Separated from this, on Fenix we have the search widget which opens the search screen. Is it considered as SAP?

**Comment by liuche** [comment]
Thanks for verifying this first piece, that we don't have those Search entry points!

The search widget should be included, so please add that SAP telemetry. Thanks for catching that!

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on  **`3.2.0-beta.3 Build #20250044 from 1/25`** using a Google Pixel 3a XL (Android 10).

## ✅ Actions

Action using the default search engine  - Ping **216ca0cf-65c3-4829-aea4-bf6ce4e50f65**
Action using one of the featured shortcuts - Ping **0acbceba-c1df-4772-9506-5d4d985d735f**
Action using a custom search engine from the shortcuts section - Ping  **844c0c3c-af38-42a6-b8fb-d89ab8023f97**
Action using as the default search engine the newly added custom search engine - Ping **32774286-f9ad-417e-97dc-fa2b9d3ab4c3**

## ✅ Suggestions
Suggestion using the default search engine  - Ping **47b8a06f-6dc7-4ff4-b0b5-e91c747fe06a**
Suggestion using one of the featured shortcuts - Ping **c2b2467c-b59e-414b-830b-a857f0a7d8fc**
Suggestion using a custom search engine from the shortcuts section - Ping **8f1bf85e-477b-4c5b-9d5d-170e8a667b57**
Suggestion using as the default search engine the newly added custom search engine - Ping **41af734c-25e9-4144-88c0-c9cc47c495ca**

## ✅ Widget
Action using the default search engine  - Ping **2a23937f-44fb-4889-8f39-0aa6402ee8f3**
Suggestion using the default search engine  - Ping **47db1e12-786d-46a2-8dc1-82365ade6594**
Action using one of the featured shortcuts - Ping **cbf9326b-9997-4720-923b-e2e834cc3023**
Suggestion using one of the featured shortcuts - Ping **1a8ca4ae-63bc-4676-bf67-eb3a9246bdd2**
Action using a custom search engine from the shortcuts section - Ping **02b4deba-8c57-42da-b6f6-1bce919e429e**
Suggestion using a custom search engine from the shortcuts section - Ping **5b648725-e256-401b-84dc-d014a6ca8dcb**
Action using as the default search engine the newly added custom search engine - Ping **af30877d-a65f-4d8b-a367-9a82b7262956**
Suggestion using as the default search engine the newly added custom search engine - Ping **2441de9b-dbff-4fe9-97d2-de7beafce5f3**

## ✅ Shortcut (Long tap on Fenix from the device's home screen)
### ✔️ Normal browsing
Action using the default search engine  - Ping **15d31611-ba3f-45ca-bbda-390286f4f970**
Suggestion using the default search engine  - Ping **72af1b10-0816-4636-8b63-045aedef3ee6**
Action using one of the featured shortcuts - Ping **a38dd735-dc13-435d-b304-8207f5a688fe**
Suggestion using one of the featured shortcuts - Ping **fec0b46d-a6f7-4405-877b-dbdd3f0b0b20**
Action using a custom search engine from the shortcuts section - Ping **c46dfd6a-98e4-46c4-9068-ef85af84a979**
Suggestion using a custom search engine from the shortcuts section - Ping **c6edbea8-0ad4-45fd-bf62-45ee666e23de**
Action using as the default search engine the newly added custom search engine - Ping **56a01c3b-c85e-44ca-bb3a-d8d7d56eda61**
Suggestion using as the default search engine the newly added custom search engine - Ping **a340a874-9b28-46fb-9e07-dbbf7c099e46**


### ✔️ Private browsing
Action using the default search engine  - Ping **2f0d7284-e1ad-4f00-8d35-142f7eb1f23a**
Suggestion using the default search engine  - Ping **d07f9737-3a33-4f72-8c33-27e3dd157f75**
Action using one of the featured shortcuts - Ping **540f71b6-7575-4e76-b6d5-88f38599914b**
Suggestion using one of the featured shortcuts - Ping **44a55ec1-dc24-4b0d-a98e-59413c2e3b79**
Action using a custom search engine from the shortcuts section - Ping **19cdeaab-9fb9-411f-a503-f02b67044d6b**
Suggestion using a custom search engine from the shortcuts section - Ping **2c8c8fa5-3866-4dc5-a31e-f988e6154b18**
Action using as the default search engine the newly added custom search engine - Ping **c8bf4098-a96d-4b63-bb0c-8d16f7d8b247**
Suggestion using as the default search engine the newly added custom search engine - Ping **42f46d38-4abf-4087-bad6-1d04b96c47c0**

## ✅ Using the added Private Browsing shortcut 
Action using the default search engine  - Ping **18d279fe-034b-481e-955e-9cfbd36add86**
Suggestion using the default search engine  - Ping **d71fded6-242a-4249-bab6-f68381b5fab2**
Action using one of the featured shortcuts - Ping **22bf20b3-8d37-471e-9730-9efb38a6a963**
Suggestion using one of the featured shortcuts - Ping **bef176cb-c187-4278-92ba-a6c643d06226**
Action using a custom search engine from the shortcuts section - Ping **bdec14a8-bfcc-4053-8567-b6854c028c40**
Suggestion using a custom search engine from the shortcuts section - Ping **6af1f7f8-cfd2-48fd-8124-ef5aff6bb7e2**
Action using as the default search engine the newly added custom search engine - Ping **35ccbb96-ed27-4850-bbe8-24aadd221419**
Suggestion using as the default search engine the newly added custom search engine - Ping **5ecffcc7-e961-4b20-902f-74d95deb6f2b**
 
## ✅ Metrics 
Metrics ping containing the search_count totals - Ping **dadbeaa7-ec12-42e8-82ff-27cfc56a7c8c** 
```
"metrics.search_count": {
            "amazondotcom.action": 2,
            "amazondotcom.shortcut": 6,
            "amazondotcom.suggestion": 2,
            "amazondotcom.widget": 2,
            "custom.action": 4,
            "custom.shortcut": 12,
            "custom.suggestion": 4,
            "custom.widget": 4,
            "google-b-1-m.action": 2,
            "google-b-1-m.shortcut": 6,
            "google-b-1-m.suggestion": 2,
            "google-b-1-m.widget": 2
          }
```

[Logcat](https://github.com/mozilla-mobile/fenix/files/4121649/Counting.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-t2801000)
